app/models/solidus_paypal_braintree/transaction_import.rb

```diff
@@ -69,10 +63,26 @@ def address
       end
     end
 
+    def state_before_current?(state)
+      steps = order.checkout_steps
+      steps.index(state) < (steps.index(order.state) || 0)
+    end
+
     protected
 
-    def advance_order(state = "confirm")
-      order.next! until order.state == state
+    def advance_order(payment, end_state)
+      return if state_before_current?(end_state)
+
+      until order.state == end_state
+        order.next!
+        update_payment_total(payment) if order.payment?
+      end
+    end
+
+    def update_payment_total(payment)
+      unless payment.amount == order.total
+        payment.update!(amount: order.total)
```

Comment:
This is likely to cause some issues in any sort of split-payment setup. (e.g.: Store credits.)